targets/charge_controller/main.cpp

```diff
@@ -174,8 +135,24 @@ int main() {
 
     // charge controller module instantiation
     BMSManager bms(can, bmsOK);
-    LCDDisplay display(LCDRegisterSEL, LCDReset, spi);
-    ChargeController chargeController(bms, display, can, startButton, statusLED);
+
+    //Model initialization
+    ControllerModel model;
+
+    //LCD display initialization
+    LCDDisplay display(LCDRegisterSEL, LCDReset, spi, model);
+
+    //Encoder GPIO pin initialization
+    IO::GPIO& encoderGPIOpinA = IO::getGPIO<ENCODER_A_PIN>(IO::GPIO::Direction::INPUT);
+    IO::GPIO& encoderGPIOpinB = IO::getGPIO<ENCODER_B_PIN>(IO::GPIO::Direction::INPUT);
+    DEV::Encoder encoder(encoderGPIOpinA, encoderGPIOpinB, 1, 0, true);
+    //Encoder button
+    IO::GPIO& encoderButtonGPIO = IO::getGPIO<ENCODER_BUTTON_PIN>(IO::GPIO::Direction::INPUT);
+    DEV::Button encoderButton(encoderButtonGPIO, IO::GPIO::State::LOW);
+
+    //UI Controller initialization
+    ControllerUI controllerUI(encoder, encoderButton, display, model);
+    ChargeController chargeController(bms, display, can, startButton, statusLED, controllerUI, model);
 
     uart.printf("Modules Initialized\n\r");
 
```

Comment:
Update the stuff below here to match our new standards for CANopen main files. Look at Taylor's documentation for guidance